libs/wingsdk/src/shared-gcp/bucket.inflight.ts

```diff
@@ -0,0 +1,257 @@
+import { Storage, Bucket } from "@google-cloud/storage";
+import {
+  BucketDeleteOptions,
+  IBucketClient,
+  ObjectMetadata,
+  BucketSignedUrlOptions,
+} from "../cloud";
+import { Json } from "../std";
+
+export class BucketClient implements IBucketClient {
+  private bucketName: string;
+  private storage: Storage;
+  private bucket: Bucket;
+
+  constructor(
+    bucketName: string,
+    storage: Storage,
+    projectId: string = process.env.GCP_PROJECT_ID || "wingsdk-test"
+  ) {
+    this.bucketName = bucketName;
+    this.storage = storage ? storage : new Storage({ projectId: projectId });
+    this.bucket = this.storage.bucket(this.bucketName);
+  }
+
+  public async metadata(_key: string): Promise<ObjectMetadata> {
+    throw new Error("Method not implemented.");
+  }
+
+  // check if bucket is public or not from bucket metadata and set it to _public
+  public async isPublic(): Promise<boolean> {
+    try {
+      const [metadata] = await this.bucket.getMetadata();
+      return metadata.iamConfiguration?.publicAccessPrevention === "inherited";
+    } catch (error) {
+      throw new Error(
+        `Failed to check if bucket is public. (bucket=${this.bucketName})`
+      );
+    }
+  }
+
+  /**
+   * Check if an object exists in the bucket
+   *
+   * @param key Key of the object
+   */
```

Comment:
@subh-cs I'm wondering if we can safely omit these docstrings for inflight clients (would this cause any build errors?)

If they are copied from `cloud/bucket.ts` then there will be a lot of duplication, and information will get out of date... (cc @tsuf239)
```suggestion
```

Reply:
I don't think these will cause any build error, and yes they are copied from `cloud/bucket.ts`. I'll removing them.